### DISCUSSION TITLE
Z-Wave JS headless

### DISCUSSION THREAD

**Discussion by AWSb24** [zwave-js/zwave-js#3165] [Q&A] [answered]
I am just migrating from OZW to Z-Wave JS. I am running OZW on two dedicated RPIs spread over my house to cover all devices. Due to concrete walls and ceilings, I am not able to cover the house with one RPI. I am running HA on an Intel server in my basement. How can install Z-Wave JS headless on the RPIs? Can I connect them with the HA integration by ws://ip-address:3000?

Thanks in advance!

**Comment by AlCalzone** [comment]
You can run the Z-Wave driver on another host than where HA is installed, but I don't know how since I don't use HA.
I suggest you ask this in the HA repo or on their discord.

**Comment by raman325** [comment]
zwave-js-server is the headless application that HA wants to talk to

The zwave-js-server README includes basic instructions on how to set it up if you want to do it on the command line, with some additional info in an open issue on that repo. What isn't covered is how to set it up as a service to auto run, or how to setup dependencies, but those can be figured out with some Googling.

There's also at least one docker image available in DockerHub to run zwave-js-server

**Comment by kpine** [comment, marked as answer]
This is [answered](https://www.home-assistant.io/integrations/zwave_js/#running-the-z-wave-js-server) in the HA docs for the integration. Either run zwavejs2mqtt (installed the way you want) or use the standalone server as Raman has described.

**Comment by AWSb24** [comment]
Thank you to you all for helping.

I tried to install it zwave-js and zwave-js-server without docker but I was not successful also with some Googling.

I ran following commands:
npm i zwave-js
npm i zwave-js-server
npm i yarn

I tried to ran ts-node src/bin/server.ts /dev/ttyACM0 but it didn't work.

What did I miss?

Thanks

**Comment by AWSb24** [comment]
Got it up and running...
